jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1488,13 +1489,33 @@ Method getMethodHelper(
 			return null;
 		}
 	}
-	result = forDeclaredMethod ? getDeclaredMethodImpl(name, parameterTypes, strSig, null) : getMethodImpl(name, parameterTypes, strSig);
+	
+	if (forDeclaredMethod) {
+		result = getDeclaredMethodImpl(name, parameterTypes, strSig, null);
+	} else {
+		result = getMethodImpl(name, parameterTypes, strSig);
+		/* Retrieve the specified method implemented by the superclass from the top to the bottom
+		 * Note: there is no need do so when the method is declared by the current class.
+		 */
+		if ((result != null) && result.getDeclaringClass().isInterface() && (this != Object.class)) {
+			if (!this.isInterface()) {
+				HashSet<Class<?>> interfaceSet = new HashSet();
```

Comment:
Any particular reason to create `new HashSet()` without actual usage after the following method call?

Reply:
If an interface inherits the same superinterface more than once somewhere in its hierarchy the HashSet will save some native calls.
example:
```interface X {void m() {}}
interface Y extends X {void m() {}}
interface Z extends X, Y {}```
X will only be searched once. this example is trivial but if X had many superinterfaces it would show an impact.